scout/parse/variant/variant.py

```diff
@@ -418,6 +424,94 @@ def set_str_info(variant: Variant, parsed_variant: Dict[str, Any]):
+        if value == "nan":
+            return ""
+        else:
+            return value
+
+    parsed_variant["gene_a"] = call_safe(str, variant.INFO.get("GENEA", ""))
+    parsed_variant["gene_b"] = call_safe(str, variant.INFO.get("GENEB", ""))
+    parsed_variant["tool_hits"] = call_safe(str, variant.INFO.get("TOOL_HITS", 0))
+    parsed_variant["fusion_score"] = call_safe(str, variant.INFO.get("SCORE", None))
+    parsed_variant["hgnc_id_a"] = call_safe(int, variant.INFO.get("HGNC_ID_A", 0))
+    parsed_variant["hgnc_id_b"] = call_safe(int, variant.INFO.get("HGNC_ID_B", 0))
+    parsed_variant["orientation"] = call_safe(str, variant.INFO.get("ORIENTATION", ""))
+    parsed_variant["frame_status"] = call_safe(
+        str, replace_nan(variant.INFO.get("FRAME_STATUS", ""))
+    )
+    parsed_variant["transcript_id_a"] = call_safe(
+        str, replace_nan(variant.INFO.get("TRANSCRIPT_ID_A", ""))
+    )
+    parsed_variant["transcript_id_b"] = call_safe(
+        str, replace_nan(variant.INFO.get("TRANSCRIPT_ID_B", ""))
+    )
+    parsed_variant["exon_number_a"] = call_safe(
+        str, call_safe(int, replace_nan(variant.INFO.get("EXON_NUMBER_A", "")))
+    )
+    parsed_variant["exon_number_b"] = call_safe(
+        str, call_safe(int, replace_nan(variant.INFO.get("EXON_NUMBER_B", "")))
+    )
+    parsed_variant["chr_a"] = call_safe(str, variant.INFO.get("CHRA", ""))
+    parsed_variant["chr_b"] = call_safe(str, variant.INFO.get("CHRB", ""))
+    parsed_variant["position_a"] = call_safe(str, variant.INFO.get("POSA", ""))
+    parsed_variant["position_b"] = call_safe(str, variant.INFO.get("POSB", ""))
+    parsed_variant["breakpoint_a"] = f"{parsed_variant['chr_a']}:{parsed_variant['position_a']}"
+    parsed_variant["breakpoint_b"] = f"{parsed_variant['chr_b']}:{parsed_variant['position_b']}"
+    parsed_variant["hgnc_symbols"] = [parsed_variant["gene_a"], parsed_variant["gene_b"]]
+
+
+def add_gene_and_transcript_info_for_fusions(parsed_variant: Dict[str, Any], variant: Variant):
+    """Add gene and transcript info for fusions. Return list of parsed
+    transcripts for later use in parsing.
+        Args:
+            parsed_variant(dict)
+            variant(cyvcf2.Variant)
+        Return:
+            parsed_transcripts(list)
+    """
+    parsed_transcripts = []
+    # If fusions have transcript information about both  fusion partners
+    if parsed_variant["transcript_id_a"] and parsed_variant["transcript_id_b"]:
+        for suffix in ["a", "b"]:
+            parsed_transcripts.append(
+                {
+                    "transcript_id": parsed_variant[f"transcript_id_{suffix}"],
+                    "hgnc_id": parsed_variant[f"hgnc_id_{suffix}"],
+                    "hgnc_symbol": parsed_variant[f"gene_{suffix}"],
+                    "exon": parsed_variant[f"exon_number_{suffix}"],
+                }
+            )
+
+    genes = []
+    for suffix in ["a", "b"]:
+        genes.append(
```

Comment:
I would have only one for loop `for suffix in ["a", "b"]:`and inside it handle both parsed stranscripts and genes. So you avoid repeated code